portal-impl/src/com/liferay/portal/systemevent/SystemEventAdvice.java

```diff
@@ -225,6 +233,32 @@ protected long getCompanyId(ClassedModel classedModel) {
 		return 0;
 	}
 
+	protected String getExternalReferenceCode(ClassedModel classedModel)
+		throws Exception {
+
+		Class<?> modelClass = classedModel.getClass();
```

Comment:
I think you could try with the `ExternalReferenceCodeResource` implementation instead of calling via reflection (maybe it would be necessary both, but I am not sure about this)

Reply:
Sorry I meant `ExternalReferenceCodeModel`!